### PR TITLE
3D UNet

### PR DESCRIPTION
**Pull request by mnabian**
# Modulus Pull Request

## Description

Closes https://github.com/NVIDIA/modulus/issues/464

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/modulus/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The [CHANGELOG.md](https://github.com/NVIDIA/modulus/blob/main/CHANGELOG.md) is up to date with these changes.
- [x] An [issue](https://github.com/NVIDIA/modulus/issues) is linked to this pull request.

### CODE REVIEW

File: modulus/models/unet/unet3d.py
Comment (ktangsali) [inline modulus/models/unet/unet3d.py:412]: Linking this issue here: https://github.com/NVIDIA/modulus/issues/512 - we should try to add support for 2D soon. In that case, should the name be `UNet` ?

File: modulus/models/unet/unet3d.py
Comment (ktangsali) [inline modulus/models/unet/unet3d.py:28]: Can the layers from here be reused? https://github.com/NVIDIA/modulus/blob/main/modulus/models/rnn/layers.py

I believe we have implementation of ConvBlock, TransposeConvBlock and ResidualBlock. Alternatively, can we use this PR to unify them?

File: modulus/models/unet/unet3d.py
Comment (ktangsali) [inline modulus/models/unet/unet3d.py:422]: There will be some correlation between the depth and the feature maps correct? Should we add some checks for that?

File: modulus/models/unet/unet3d.py
Comment (ktangsali) [inline modulus/models/unet/unet3d.py:424]: Would be good to list the options

File: modulus/models/unet/unet3d.py
Comment (ktangsali) [inline modulus/models/unet/unet3d.py:432]: Would be good to expose the different activation functions here

File: test/models/test_unet3d.py
Comment (ktangsali) [inline test/models/test_unet3d.py:1]: would be great to parameterize these tests for:
1. multiple depths / feature maps
2. conv blocs num
3. input / output channels

File: modulus/models/unet/unet3d.py
Comment (mnabian) [reply modulus/models/unet/unet3d.py:412]: Renamed to UNet. Seems like we have a volunteer to add support for 2D after this PR is merged: https://github.com/NVIDIA/modulus/issues/512#issuecomment-2138534646

File: modulus/models/unet/unet3d.py
Comment (mnabian) [reply modulus/models/unet/unet3d.py:422]: There are already checks in the encoder and decoder for that:

https://github.com/mnabian/modulus/blob/fea-ext-3dunet/modulus/models/unet/unet3d.py#L293

https://github.com/mnabian/modulus/blob/fea-ext-3dunet/modulus/models/unet/unet3d.py#L349

File: modulus/models/unet/unet3d.py
Comment (mnabian) [reply modulus/models/unet/unet3d.py:424]: Added

File: modulus/models/unet/unet3d.py
Comment (mnabian) [reply modulus/models/unet/unet3d.py:28]: Well, not exactly. The conv block in this PR also includes normalization and activation, which is not there in the rnn layers. I agree that we have different implementation of these layers, and that eventually these have to be unified. For now, hopefully, the api documentation helps to distinguish between these layers/modules.

File: test/models/test_unet3d.py
Comment (mnabian) [reply test/models/test_unet3d.py:1]: Added these to the constructor test

**Review by loliverhennigh — APPROVED** [review]
Ill take another look tomorrow morning but used this model a few times with success. Approving now in case it holds up release stuff.

File: modulus/models/unet/unet3d.py
Comment (ktangsali) [reply modulus/models/unet/unet3d.py]: Sounds good. Let's make an issue for this, so that it can be tracked in the future

File: modulus/models/unet/unet3d.py
Comment (ktangsali) [reply modulus/models/unet/unet3d.py]: Awesome, thanks!

File: modulus/models/unet/unet3d.py
Comment (mnabian) [reply modulus/models/unet/unet3d.py]: Sure, I added it to this list: https://github.com/NVIDIA/modulus/issues/463

**Review by ktangsali — APPROVED** [review]
LGTM!

**Pull request merged**
